### ISSUE TITLE
Find a way to use /dev/shm to store the lmod cache for speed up; need a healthcheck

### ISSUE THREAD

**Issue by fgeorgatos** [plabrop/bic#49]
per @rtmclay , this would benefit from /etc/lmodrc.lua or so

**Comment by fgeorgatos** [comment]
OK, this seems to have required an incantation of the style (`-t` proved critical):
* `$LMOD_DIR/update_lmod_system_cache_files -d /dev/shm/lmod/cacheDir -t /dev/shm/lmod/system.txt $MODULEPATHLIST`

once that is done right, it works as intended:

```
[fgeorgatos@node002 ~]$ ml av -D R/3.4 2>&1 |grep R.3.4|sed 's/\/.*HPCBIOS/...HPCBIOS/g'
lmod(-D avail R/3.4){
  Master:avail(R/3.4){
    ["...HPCBIOS/20170505/el7/0.1.0-86-gdec7b3e/modules/all/R/3.4.0-intel-2017a-X11-20170314"]  = {
      ["fullName"] = "R/3.4.0-intel-2017a-X11-20170314",
        ["...HPCBIOS/20170505/el7/0.1.0-86-gdec7b3e/software/R/3.4.0-intel-2017a-X11-20170314/lib64"] = 1,
        ["...HPCBIOS/20170505/el7/0.1.0-86-gdec7b3e/software/R/3.4.0-intel-2017a-X11-20170314/lib64/R/lib"] = 1,
        ["...HPCBIOS/20170505/el7/0.1.0-86-gdec7b3e/software/R/3.4.0-intel-2017a-X11-20170314/bin"] = 1,
          icnt: 260, fullName: R/3.4.0-intel-2017a-X11-20170314
   R/3.4.0-intel-2017a-X11-20170314 (D)
[fgeorgatos@node002 ~]$ ml future HPCBIOS/2017q2-el7
[fgeorgatos@node002 ~]$ time ml av R/3.4

--------------------- /...HPCBIOS/20170505/el7/0.1.0-86-gdec7b3e/modules/all ---------------------

   R/3.4.0-intel-2017a-X11-20170314 (D)
[...]
real	0m1.055s
user	0m0.200s
sys	0m0.045s
[fgeorgatos@node002 ~]$ ll .lmod.d/.cache/spiderT.x86_64_Linux.luac_5.1
ls: cannot access .lmod.d/.cache/spiderT.x86_64_Linux.luac_5.1: No such file or directory
```

as opposed to:
```
[fgeorgatos@node002 ~]$ time ml --ignore_cache av R/3.4

--------------------- /...HPCBIOS/20170505/el7/0.1.0-86-gdec7b3e/modules/all --------------------- 
   R/3.4.0-intel-2017a-X11-20170314 (D)
[...]
real	1m16.098s
user	0m42.773s
sys	0m25.702s
```

**Comment by fgeorgatos** [comment]
interestingly, that last `--ignore_cache` command, creates the caches regardless; imho, it may be a bug?:
```
[fgeorgatos@node002 ~]$ ll .lmod.d/.cache/spiderT.x86_64_Linux.luac_5.1
-rw-r--r-- 1 fgeorgatos Forest_Users 3390552 Jun 13 00:03 .lmod.d/.cache/spiderT.x86_64_Linux.luac_5.1
```

**Comment by rtmclay** [comment]
This has been fixed with Lmod 7.4.21
